### PR TITLE
improved color editor, remove unused DataUi

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Removed DataUi (taken over by edit ui instead):
* `Range1D`
* `Material`
* `Color`

Improved Color editor ui (more similar to what color DataUI was like before) and use it for mesh material as well.

* Mesh material editing suffers from https://github.com/rerun-io/rerun/issues/6624

Before:
image

After:
image
image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6645?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6645?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6645)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_edit_ui/src/color.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    *value = edit_color.into();
+
+    ui.painter().rect_stroke(
+        response.rect,
+        1.0,
+        ui.visuals().widgets.noninteractive.fg_stroke,
+    );
+
+    let [r, g, b, a] = edit_color.to_array();
+    response.on_hover_text(format!("Color #{r:02x}{g:02x}{b:02x}{a:02x}"))
```
Comment (abey79) [inline crates/re_edit_ui/src/color.rs:26]: Might be worth adding `on_disabled_hover_text` here as well. I'm expecting that the behaviour for e.g. the "Fallback" line would be:
- hover on "Fallback" text -> tooltip explains what it is
- hover on Fallback's colour widget -> tooltip gives colour value
...which is kind of nice.

File: crates/re_edit_ui/src/material.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+use re_types::{components::Material, datatypes::Rgba32};
+use re_viewer_context::ViewerContext;
+
+use crate::color::edit_color_ui;
+
+// TODO(andreas): as we add more elements to the Material struct, we'll need a multi-line editor.
+//                  begs the question though if a material struct is really the right way to go!
+pub fn edit_material_ui(
```
Comment (abey79) [inline crates/re_edit_ui/src/material.rs:8]: Related (not this PR): https://github.com/rerun-io/rerun/issues/6624#issuecomment-2191158915

**Review by abey79 — APPROVED** [review]
I like it when both the UI and the code improve at the same time :)

File: crates/re_edit_ui/src/color.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_edit_ui/src/color.rs:26]: oh I thought `on_hover_text` is triggered even when it's disabled. I have to set the same for `on_disabled_hover_text`?

Comment (abey79) [reply crates/re_edit_ui/src/color.rs:26]: Actually I'm not entirely sure. But currently it doesn't trig for the "Fallback" item. Maybe that's another issue?

**Pull request merged**